cloudify_cli/commands/agents.py

```diff
@@ -0,0 +1,137 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import threading
+
+from cloudify_cli import utils
+from cloudify_cli.logger import get_logger
+from cloudify_cli.execution_events_fetcher import wait_for_execution
+from cloudify_cli.exceptions import SuppressedCloudifyCliError
+from cloudify_cli.exceptions import ExecutionTimeoutError
+from cloudify import logs
+
+_NODE_INSTANCE_STATE_STARTED = 'started'
+
+
+def _is_deployment_installed(client, deployment_id):
+    for node_instance in client.node_instances.list(deployment_id):
+        if node_instance.state != _NODE_INSTANCE_STATE_STARTED:
```

Comment:
@idanmo Just making sure that we know that this will return False if we're in the middle of an install workflow. Since this practically checks for an arbitrary node's state.